src/direct/proof.rs

```diff
@@ -0,0 +1,83 @@
+use ark_crypto_primitives::{
+    merkle_tree::{Config as MerkleConfig, LeafParam, MultiPath, TwoToOneParam},
+    sponge::{Absorb, CryptographicSponge},
+};
+use ark_ff::FftField;
+use ark_std::vec::Vec;
+
+use crate::utils::squeeze_integer;
+
+pub struct DirectProof<F, M, S>
+where
+    F: FftField,
+    M: MerkleConfig,
+    S: CryptographicSponge,
+{
+    challenge_answers: MultiPath<M>,
+    committed_values: Vec<Vec<F>>,
+    merkle_leaf_hash_param: LeafParam<M>,
```

Comment:
How would you verify if they're not in the proof? MultiProof().verify() takes these as arguments

Reply:
The verifier will have them in its own config no? So it can pass them to this.

Reply:
It doesn't bother me here and I think it would be weird to "pass it in" as an argument to the proof.verify(..., merkle_params): https://github.com/arkworks-rs/ldt/blob/z-tech/STIR/src/direct/proof.rs#L65, but if we delete proof.verify() entirely and move it onto the DirectProver implementation that seems fine to me